mapper/orm-coordination-outbox-polling/src/main/java/org/hibernate/search/mapper/orm/coordination/outboxpolling/cluster/impl/OutboxPollingAgentAdditionalJaxbMappingProducer.java

```diff
@@ -168,38 +132,55 @@ public Collection<MappingDocument> produceMappings(ConfigurationPropertySource p
 					ENTITY_MAPPING_AGENT_PAYLOAD_TYPE.resolveOrRaw( propertySource ) );
 		}
 
-		String resolvedUuidType = UuidDataTypeUtils.uuidType(
-				uuidType.orElse( HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_UUID_TYPE ),
-				dialect );
-
-		@SuppressWarnings("deprecation")
-		String entityDefinition = mapping.orElseGet( () -> String.format(
-				Locale.ROOT,
-				ENTITY_DEFINITION_TEMPLATE,
-				schema.orElse( "" ),
-				catalog.orElse( "" ),
-				table.orElse( HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_TABLE ),
-				uuidStrategy.orElse(
-						HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_UUID_GEN_STRATEGY )
-						.strategy(),
-				resolvedUuidType,
-				PayloadMappingUtils.payload(
-						payloadType.orElse(
-								HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_PAYLOAD_TYPE ),
-						true
-				)
-		) );
-
-		log.agentGeneratedEntityMapping( entityDefinition );
-		Origin origin = new Origin( SourceType.OTHER, "search" );
-
-		ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream( entityDefinition.getBytes() );
-		BufferedInputStream bufferedInputStream = new BufferedInputStream( byteArrayInputStream );
-		Binding<?> binding = mappingBinder.bind( bufferedInputStream, origin );
-
-		JaxbHbmHibernateMapping root = (JaxbHbmHibernateMapping) binding.getRoot();
-
-		MappingDocument mappingDocument = new MappingDocument( HIBERNATE_SEARCH, root, origin, buildingContext );
-		return Collections.singletonList( mappingDocument );
+		JaxbEntityMappings mappings;
+		if ( mapping.isPresent() ) {
+			mappings = unmarshall( mapping.get() );
+		}
+		else {
+			int resolvedUuidType = UuidDataTypeUtils.uuidType(
+					uuidType.orElse(
+							HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_UUID_TYPE ),
+					dialect );
+			@SuppressWarnings("deprecation")
+			int resolvedPayloadType = PayloadMappingUtils.payload(
+					payloadType.orElse(
+							HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_PAYLOAD_TYPE )
+			);
+			String resolvedUuidStrategy = uuidStrategy.orElse(
+					HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_UUID_GEN_STRATEGY )
+					.strategy();
+
+			mappings = createMappings(
+					schema.orElse( "" ),
+					catalog.orElse( "" ),
+					table.orElse( HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_TABLE ),
+					resolvedUuidType, resolvedPayloadType, resolvedUuidStrategy,
+					buildingContext.getMetadataCollector().getFilterDefinition( TenantIdBinder.FILTER_NAME ) != null
+			);
+		}
+
+		log.agentGeneratedEntityMapping( marshall( mappings ) );
```

Comment:
You should use a formatter in the definition of the logging method, so that `marshall` is only called when this is actually logged (in this case, when debug level is enabled, IIRC).

And if that's too complicated, then `mappings.toString()` may be good enough for what we're doing here (debugging).